### PR TITLE
Markdown support

### PR DESCRIPTION
**Pull request by juanedi**
## Markdown support

For #34. This PR adds support for verifying examples in markdown files. Some non-trivial refactoring was involved, so here's a summarized overview of the biggest changes/decisions:

### Organization

Format-specific stuff (parsing, naming conventions, etc.) was moved to either `VerifyExamples.Elm` or `VerifyExamples.Markdown` and their submodules.

### Snippets

The `Parser` module now receives a `snippet` instead of a `comment`. A snippet is the source code inside a comment.

Before this change, we were grabbing all the content inside an elm comment and passing that to the parser, which would discard lines that didn't have 4 spaces of indentation (that was done in `VerifyExamples.Ast.Intermediate`).

Now the parser receives clean snippets of code, and it's up to each input format to extract those from comments.

### Warnings

Markdown input files don't support warnings. At this point it wouldn't be too hard to leave the code ready for them, but I couldn't come up with a very good use case for that.

### elm-verify-examples.json

Besides elm module names, the configuration now supports relative paths to markdown files. This feels a little inconsistent, but felt more pragmatic than using two separate lists (`elmTests`/`markdownTests`).

Maybe we should use file paths for all of them?

### CODE REVIEW

**Comment by juanedi** [comment]
@stoeffel mind taking a look? 😄 
Left a brief overview in the PR description above, but happy to give more context if the diff is too big.

**Comment by stoeffel** [comment]
@juanedi just made a first pass and will try it on my machine, but on a first glance this looks great!

File: Readme.md [hunk 1]
```diff
@@ -31,7 +31,8 @@ $ touch tests/elm-verify-examples.json
   "root": "../src",
   "tests": [
     "Mock",
-    "Mock.Foo.Bar.Moo"
+    "Mock.Foo.Bar.Moo",
+    "./README.md"
```
Comment (stoeffel) [inline Readme.md]: 💭 I wonder if this should be relative from `"root"` above. It's unclear to me where this is relative to.
We probably also want to support `../README.md`.
@juanedi what do you think?

Comment (stoeffel) [reply Readme.md]: also it fails with `content` isn't defined on `data`. If it doesn't find a file.

**Comment by stoeffel** [comment]
started a branch based on this to change some stuff.

File: Readme.md (hunk 1 above)
Comment (stoeffel) [reply Readme.md]: nope, reconsidered. this is okay.

**Pull request merged**